WebService/v1/Models/SimulationApiModel/SimulationApiModel.cs

```diff
@@ -165,7 +220,92 @@ public async Task ValidateInputRequest(ILogger log, IIotHubConnectionStringManag
                 throw new BadRequestException(INVALID_DATE, e);
             }
 
-            await connectionStringManager.ValidateConnectionStringAsync(this.IotHub.ConnectionString);
+            foreach (var iotHub in this.IotHubs)
+            {
+                await connectionStringManager.ValidateConnectionStringAsync(iotHub.ConnectionString);
+            }
+        }
+
+        // Append additional Hub properties and Statistics 
+        private void AppendHubPropertiesAndStatistics(
+            IServicesConfig servicesConfig, 
+            IDeploymentConfig deploymentConfig, 
+            IIotHubConnectionStringManager connectionStringManager,
+            ISimulationRunner simulationRunner,
+            IRateLimiting rateReporter)
+        {
+            var isRunning = this.Running == true;
+
+            foreach (var iotHub in this.IotHubs)
+            {
+                // Preprovisioned hub status
+                var isHubPreprovisioned = this.IsHubConnectionStringConfigured(servicesConfig);
+
+                if (isHubPreprovisioned && isRunning)
+                {
+                    iotHub.PreprovisionedIoTHubInUse = this.IsPreprovisionedIoTHubInUse(servicesConfig, connectionStringManager);
+                    var preprovisionedIoTHubMetricsUrl = this.GetIoTHubMetricsUrl(servicesConfig, deploymentConfig, connectionStringManager);
+                    iotHub.PreprovisionedIoTHubMetricsUrl = preprovisionedIoTHubMetricsUrl;
+                }
+            }
+
+            if (isRunning)
+            {
+                // Average messages per second count
```

Comment:
nit: "Average messages per second count" => "Average messages per second frequency in the last minutes"

Reply:
:white_check_mark: